### PR TITLE
Fixes the wrong fire being applied after hit by plasma sentry

### LINKED ISSUE #2985: Plasma Sentry - Incindiary OB bug
### Testmerges

N/A - Tested locally

### Description of the bug

If a burrower is hit by an incendiary OB while burrowed they won't be set on fire (as is intended), however when a plasma sentry hits them afterwards the fire of the OB will get set to them instead of the plasma sentry's.

https://user-images.githubusercontent.com/15560820/229170431-4e563091-1d57-4053-a11b-51fc8df242f3.mp4

### What's the difference with what should have happened?

Plasma sentry should apply its own fire instead of the OB's.

### How do we reproduce this bug?

1. Inc OB at Burrower.
2. Burrower burrows before OB hits.
3. Some time later.
4. Burrower gets hit by plasma sentry.
5. Unexpectedly spicy burrower barbecue.

### Issue Bingo

- [X] Issue could be reproduced at least once
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] Couldn't find an existing issue about this (https://github.com/cmss13-devs/cmss13/issues)

### PR DESCRIPTION
**Pull request by realkhad**
# About the pull request
Gives the phosphorus smoke that the plasma sentry uses a fire_reagent, fixes #2985 
# Explain why it's good for the game
fix
# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:Khadd
fix: wrong fire no longer applied
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/flamer/flamer.dm [hunk 1]
```diff
@@ -505,6 +505,11 @@ GLOBAL_LIST_EMPTY(flamer_particles)
 		if(ignited_morb.stat == DEAD)
 			continue
 
+		if(isxeno(ignited_morb))
```
[the remark this answers is not in the document]
Comment (harryob) [reply code/modules/projectiles/guns/flamer/flamer.dm:508]: that's the name of the existing var

[the remark this answers is not in the document]
Comment (realkhad) [reply code/modules/projectiles/guns/flamer/flamer.dm:508]: thats how the variable is called, someone misspelled it. can change if need-be

**Comment by realkhad** [comment]
smoke still uses previous fire_reagent, changed in 6fe0718726ef694fa3e4bae9db9545a4c9070999 to just use napalm.

**Comment by usnpeepoo** [comment]
Burrowers taking fire damage while burrowed is a balance factor against their ability, they used to be flammable while burrowed but no longer are. Removing this factor is a huge buff to the caste.

**Comment by realkhad** [comment]
> Burrowers taking fire damage while burrowed is a balance factor against their ability, they used to be flammable while burrowed but no longer are. Removing this factor is a huge buff to the caste.

fair point, i didnt realize how much of a sizable buff it'd be. i got carried away while looking at the code and thought it was unintentional.

**Review by morrowwolf — APPROVED** [review]

**Pull request merged**